src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -76,7 +88,59 @@ public static string Encode(byte[] inArray, int offset, int length)
             if (inArray == null)
                 throw LogHelper.LogArgumentNullException("inArray");
 
-            return EncodeString(Convert.ToBase64String(inArray, offset, length));
+            int i;
+
+            char[] table = s_base64Table;
+
```

Comment:
nit: remove double blank line

Reply:
Even though it seems to be a hassle, we always add an error id to Token.LogMessages so that each error has an identifier that we can link to. In this case its not that important, but its a good habit.
We could even go so far as having different message IDs for different errors.
I have rarely seen a user complain about having very distinct error messages.